### PR TITLE
Add http3 test quic interop test

### PR DESCRIPTION
**Pull request by nhorman**
Now that we have an http3 demo server, lets add support for this test into our interop container

The addition of this test incorporates our newly added ossl-nghttp3-demo-server.c code to run as a server instance against all the clients we test against.

- [x] tests are added or updated

### CODE REVIEW

File: demos/http3/ossl-nghttp3-demo-server.c [hunk 1]
```diff
@@ -84,14 +84,18 @@ static void init_ids(struct h3ssl *h3ssl)
 {
     struct ssl_id *ssl_ids;
     int i;
+    char *prior_fileprefix = h3ssl->fileprefix;
```
Comment (Sashan) [inline demos/http3/ossl-nghttp3-demo-server.c:87]: can you add this oneliner tweak here?
```
diff --git a/demos/http3/ossl-nghttp3-demo-server.c b/demos/http3/ossl-nghttp3-demo-server.c
index 9e5f8bbdb6..b4aba0ed41 100644
--- a/demos/http3/ossl-nghttp3-demo-server.c
+++ b/demos/http3/ossl-nghttp3-demo-server.c
@@ -86,6 +86,9 @@ static void init_ids(struct h3ssl *h3ssl)
     int i;
     char *prior_fileprefix = h3ssl->fileprefix;
 
+    if (h3ssl->ptr_data != NULL && h3ssl->ptr_data != nulldata)
+        free(h3ssl->ptr_data);
+
     memset(h3ssl, 0, sizeof(struct h3ssl));
 
     ssl_ids = h3ssl->ssl_ids;
```

**Review by Sashan — CHANGES_REQUESTED** [review]
While you are there...

I think `init_ids()` may leak memory, this bug is present there. can you fix it in this PR?

thanks.

**Review by t8m — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

File: demos/http3/ossl-nghttp3-demo-server.c [hunk 2]
```diff
@@ -344,9 +351,18 @@ static int quic_server_read(nghttp3_conn *h3conn, SSL *stream, uint64_t id, stru
         fprintf(stderr, "SSL_read %d on %llu failed\n",
                 SSL_get_error(stream, ret),
                 (unsigned long long) id);
-        if (SSL_get_error(stream, ret) == SSL_ERROR_WANT_READ)
-            return 0; /* retry we need more data */
-        ERR_print_errors_fp(stderr);
+        switch (SSL_get_error(stream, ret)) {
+        case SSL_ERROR_WANT_READ:
+            return 0;
+            break;
```
Comment (andrewkdinh) [inline demos/http3/ossl-nghttp3-demo-server.c:357]: Can we remove the `break` statements in these cases?

File: test/quic-openssl-docker/Dockerfile [hunk 3]
```diff
@@ -28,9 +28,10 @@ RUN git clone --depth 1 https://github.com/ngtcp2/nghttp3.git && \
 # download and build openssl 
 RUN git clone --depth 1 -b $OPENSSL_BRANCH $OPENSSL_URL && \
     cd openssl && \
-    ./Configure enable-sslkeylog enable-fips enable-demos disable-docs --prefix=/usr --openssldir=/etc/pki/tls && \
+    ./Configure enable-sslkeylog enable-fips enable-demos enable-h3demo disable-docs --prefix=/usr --openssldir=/etc/pki/tls && \
```
Comment (andrewkdinh) [inline test/quic-openssl-docker/Dockerfile:31]: Why is this not included within `enable-demos`?

**Review by shahsb — APPROVED** [review]

File: test/quic-openssl-docker/Dockerfile (hunk 3 above)
Comment (t8m) [reply test/quic-openssl-docker/Dockerfile:31]: Because it requires additional libraries to be built.

**Review by t8m — APPROVED** [review]
Still OK, I do not think this requires restarting the 24h period.

**Comment by t8m** [comment]
Partially squashed and merged to the feature branch. Thank you.

**Pull request closed without merging**